Tests/SwiftParserTest/translated/AsyncTests.swift

```diff
@@ -159,31 +141,48 @@ final class AsyncTests: XCTestCase {
   }
 
   func testAsync10c() {
+    AssertParse(
+      "typealias AsyncFunc3 = () throws 1️⃣async -> ()",
+      diagnostics: [
+        DiagnosticSpec(message: "'async' must precede 'throws'")
+      ],
+      fixedSource: "typealias AsyncFunc3 = () async throws -> ()"
+    )
+  }
+
+  func testAsync11a() {
     AssertParse(
       """
-      typealias AsyncFunc3 = () throws async -> ()
-      """,
+      let _ = [() async -> ()]()
+      """
+    )
+  }
+
+  func testAsync11b() {
+    AssertParse(
+      """
+      let _ = [() async throws -> ()]()
+      """
+    )
+  }
+
+  func testAsync11c() {
+    AssertParse(
+      "let _ = [() throws 1️⃣async -> ()]()",
       diagnostics: [
-        // TODO: Old parser expected error on line 1: 'async' must precede 'throws', Fix-It replacements: 34 - 40 = '', 27 - 27 = 'async '
-      ]
+        DiagnosticSpec(message: "'async' must precede 'throws'")
+      ],
+      fixedSource: "let _ = [() async throws -> ()]()"
     )
   }
 
-  func testAsync11() {
+  func testAsync11d() {
     AssertParse(
       """
-      // Parsing type expressions with 'async'.
-      func testTypeExprs() {
-        let _ = [() async -> ()]()
-        let _ = [() async throws -> ()]()
-        let _ = [() throws 1️⃣async -> ()]()
-        let _ = [() -> 2️⃣async ()]()
-      }
+      let _ = [() -> 1️⃣async ()]()
       """,
       diagnostics: [
-        // TODO: Old parser expected error on line 5: 'async' must precede 'throws', Fix-It replacements: 22 - 28 = '', 15 - 15 = 'async '
-        DiagnosticSpec(locationMarker: "1️⃣", message: "unexpected code 'async' in array element"),
-        DiagnosticSpec(locationMarker: "2️⃣", message: "'async' may only occur before '->'"),
+        DiagnosticSpec(message: "'async' may only occur before '->'")
```

Comment:
Do you prefer `may only occur before` vs `must precede`? I sort of like the latter because it's so succinct 😅

Reply:
I think I just copied the old parser’s error messages. I can change this this to `must precede`